spark/src/main/java/org/apache/iceberg/actions/RewriteDataFilesAction.java

```diff
@@ -111,171 +49,12 @@ protected RewriteDataFilesAction self() {
-    if (filteredGroupedTasks.isEmpty()) {
-      return RewriteDataFilesActionResult.empty();
-    }
-
-    // Split and combine tasks under each partition
-    List<CombinedScanTask> combinedScanTasks = filteredGroupedTasks.values().stream()
-        .map(scanTasks -> {
-          CloseableIterable<FileScanTask> splitTasks = TableScanUtil.splitFiles(
-              CloseableIterable.withNoopClose(scanTasks), targetSizeInBytes);
-          return TableScanUtil.planTasks(splitTasks, targetSizeInBytes, splitLookback, splitOpenFileCost);
-        })
-        .flatMap(Streams::stream)
-        .collect(Collectors.toList());
-
+  protected List<DataFile> rewriteDataForTasks(List<CombinedScanTask> combinedScanTasks) {
     JavaRDD<CombinedScanTask> taskRDD = sparkContext.parallelize(combinedScanTasks, combinedScanTasks.size());
-
-    Broadcast<FileIO> io = sparkContext.broadcast(fileIO);
-    Broadcast<EncryptionManager> encryption = sparkContext.broadcast(encryptionManager);
-
-    RowDataRewriter rowDataRewriter = new RowDataRewriter(table, spec, caseSensitive, io, encryption);
-
-    List<DataFile> addedDataFiles = rowDataRewriter.rewriteDataForTasks(taskRDD);
-    List<DataFile> currentDataFiles = filteredGroupedTasks.values().stream()
-        .flatMap(tasks -> tasks.stream().map(FileScanTask::file))
-        .collect(Collectors.toList());
-    replaceDataFiles(currentDataFiles, addedDataFiles);
-
-    return new RewriteDataFilesActionResult(currentDataFiles, addedDataFiles);
-  }
-
-  private Map<StructLikeWrapper, Collection<FileScanTask>> groupTasksByPartition(
-      CloseableIterator<FileScanTask> tasksIter) {
-    ListMultimap<StructLikeWrapper, FileScanTask> tasksGroupedByPartition = Multimaps.newListMultimap(
-        Maps.newHashMap(), Lists::newArrayList);
-
-    try {
-      tasksIter.forEachRemaining(task -> {
-        StructLikeWrapper structLike = StructLikeWrapper.forType(spec.partitionType()).set(task.file().partition());
-        tasksGroupedByPartition.put(structLike, task);
-      });
-
-    } finally {
-      try {
-        tasksIter.close();
-      } catch (IOException ioe) {
-        LOG.warn("Failed to close task iterator", ioe);
-      }
-    }
-
-    return tasksGroupedByPartition.asMap();
-  }
-
-  private void replaceDataFiles(Iterable<DataFile> deletedDataFiles, Iterable<DataFile> addedDataFiles) {
-    try {
-      RewriteFiles rewriteFiles = table.newRewrite();
-      rewriteFiles.rewriteFiles(Sets.newHashSet(deletedDataFiles), Sets.newHashSet(addedDataFiles));
-      commit(rewriteFiles);
-
-    } catch (Exception e) {
-      Tasks.foreach(Iterables.transform(addedDataFiles, f -> f.path().toString()))
-          .noRetry()
-          .suppressFailureWhenFinished()
-          .onFailure((location, exc) -> LOG.warn("Failed to delete: {}", location, exc))
-          .run(fileIO::deleteFile);
-
-      throw e;
-    }
+    Broadcast<FileIO> io = sparkContext.broadcast(SparkUtil.serializableFileIO(this.table()));
```

Comment:
nit:  we usually use `this` to assign value to a local field, so that we could distinguish it's a local field member assignment or normal field assignment.  If call the private or protected methods, we don't use `this`.